tdb.go

```diff
@@ -251,24 +251,41 @@ func (db *TrailDB) GetTrailID(cookie string) (uint64, error) {
 	return uint64(trail_id), nil
 }
 
-func (db *TrailDB) Version() int {
-	return int(C.tdb_version(db.db))
+func (db *TrailDB) Version() uint64 {
+	return uint64(C.tdb_version(db.db))
 }
 
 func (db *TrailDB) Close() {
 	C.tdb_close(db.db)
 }
 
-func NewTrail(db *TrailDB, trail_id int) (*Trail, error) {
-	trail := C.tdb_cursor_new(db.db)
-	err := C.tdb_get_trail(trail, C.uint64_t(trail_id))
+func NewCursor(db *TrailDB) (*Trail, error) {
+	var err error = nil
+    var t *Trail = nil
```

Comment:
the code doesn't compile without these initializations. The values are returned a few lines down